devolutions-session/src/dvc/channel.rs

```diff
@@ -0,0 +1,79 @@
+use std::fmt::Debug;
+
+use tokio::sync::mpsc::{self, Receiver, Sender};
+use win_api_wrappers::event::Event;
+use windows::Win32::Foundation::HANDLE;
+use windows::Win32::System::Threading::SetEvent;
+
+const IO_CHANNEL_SIZE: usize = 100;
+
+/// Mpsc channel with WinAPI event signaling.
+#[derive(Debug, Clone)]
+pub struct WinapiSignaledSender<T> {
+    tx: Sender<T>,
+    event: Event,
+}
+
+impl<T: Send + Sync + Debug + 'static> WinapiSignaledSender<T> {
+    pub async fn send(&self, message: T) -> anyhow::Result<()> {
+        self.tx.send(message).await?;
+
+        // DVC IO loop is controlled by WinAPI events signaling, therefore we need to fire event to
+        // notify DVC IO loop about new incoming message.
+
+        // SAFETY: No preconditions.
+        unsafe {
+            SetEvent(self.event.raw())?;
+        }
```

Comment:
suggestion: What about adding a method `set` to our Event wrapper?

Reply:
This was missed during refactoring, thanks, I'll revisit `SetEvent` calls